src/models/machine_learning/MLmodel.jl

```diff
@@ -7,9 +7,12 @@ _inputs_Y_law = (; T=iTemp(), H̄=iH̄())
 _inputs_U_law = (; H̄=iH̄(), ∇S=i∇S())
 
 # Abstract type as a parent type for Machine Learning models
-abstract type MLmodel <: AbstractModel end
+abstract type OptimizableModel <: AbstractModel end
```

Comment:
These should be documented somewhere, it's not 100% intuitive from the names.

Reply:
mmm I don't like much the name `OptimizableModel`, it just does not sound as a "model with parameters that can be optimized" but more as "A model that can be optimized, in sence of efficiency". Maybe `TrainableModel` or `ParametricModel`? To make emphasis that you are optimizing the parameters of the model?